### DISCUSSION TITLE
Error When Pulling Docker Image: "invalid tar header"

### DISCUSSION THREAD

**Discussion by orange-hou** [moby/moby#46491] [Q&A]
Hello everyone,

I've encountered a peculiar issue when using Docker to pull images and I'm hoping to get some help here. The problem is that sometimes when I run the docker pull command, I encounter the following error:

<img width="942" alt="63a2778fd65f210181bce291bffeb70d" src="https://github.com/moby/moby/assets/46957428/18162d64-f264-42ba-b220-29f0ce300388">
This issue occurs sporadically, and I've confirmed the following points:

1.  image itself is not the problem, as it can be pulled successfully at other times.
2. My network connection is stable, and there are no apparent network issues.
3. I've also verified that the Docker root-dir has sufficient space.

Docker: 23.0.4
OS: CentOS 7.6
Kernel: 5.10.0-1.0.0.25

It's worth noting that the problem is not consistent; extracting the image sometimes succeeds and other times fails
This Docker version works flawlessly on other machines, and I can successfully pull images there. This leads me to believe that the problem is specific to this machine's environment. Unfortunately, I'm unable to pinpoint the exact cause of the issue.
Thank you all for your assistance!

**Comment by thaJeztah** [comment]
```
Docker: 23.0.4
OS: CentOS 7.6
Kernel: 5.10.0-1.0.0.25
```

- Are you able to post the full output of `docker version` and `docker info`?
- Is there anything in your setup? (Filesystems, custom daemon configuration)
- I also notice `Kernel: 5.10.0-1.0.0.25`: that looks to be a custom kernel, correct? We generally only test / "support" the distro's kernel; this can be more important for CentOS and RHEL, as those kernels contain a lot of patches that are specific to their distro.

**Reply by orange-hou** [reply]
Thank you for your response!

- docker info output:
<img width="524" alt="8c71c4c4559a6df9dc452f2465a2beec" src="https://github.com/moby/moby/assets/46957428/84a6dc90-a165-4e14-ac15-62f932df9eaf">

- setup:
    daemon.json: only add a mirror registry and configure the data-root, enable debug
    the data-root has been set to /home/docker-overlay, here is the filesystem of the data-root:
   
<img width="690" alt="image" src="https://github.com/moby/moby/assets/46957428/679ed9f3-627e-4eea-bbb9-1dbe12129240">

- yeh, it's a custom kernel, I'm not sure if it's a kernel-related issue, so I will try reinstalling a different version of the kernel to test it
- Additionally, i found the msg in the Docker log during Docker startup, this msg also appear on other machines that can successfully pull images:
 `No quota support for local volumes in /home/docker-overlay/volumes: Filesystem does not support, or has not enabled quotas`
